### ISSUE TITLE
AMD execution function is not supplied with the proper amount of arguments

### ISSUE THREAD

**Issue by KinsheX** [systemjs/systemjs#1932]
Hello,

I have recently started looking into SystemJS, and the different module formats it can load. However, I have encountered a bit of a hurdle when trying to load in a UMD bundle generated by the Angular ng-packagr.

The top of the UMD bundle looks like this:
```
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('@app/api', ['exports', '@angular/common', '@angular/core'], factory) :
    (factory((global['app'] = global['app'] || {}, global['app']['api'] = {}),global.ng.common,global.ng.core));
}(this, (function (exports,common,core) { 'use strict';
```
The bundle is loaded with `SystemJS.import()` and fails because "core" is undefined. When debugging I can see that the "exports" and "common" variables have received the two actual dependencies, such that:

```
exports -> @angular/common
common -> @angular/core
core -> undefined
```

I can see in the `extras/amd.js` that the `createAMDRegister` function is slicing the "exports" part off when calling the `amdExec` function.

I would assume that the amdExec function signature matches that which is defined in the AMD `define`.

Version: 3.1.2

**Comment by guybedford** [comment]
I've just updated the test case for this (https://github.com/systemjs/systemjs/commit/af700ef99ea390a4d7eee483f1df6c888f1230a0) and can confirm the issue is unrelated to the 'exports' thing.

Are you sure you're getting those offsets?

I'm wondering if this might not be an interop issue that you are seeing undefined rather.

Note that dependencies of AMD modules, must be provided as `default` exports.

**Comment by KinsheX** [comment]
During debugging it seems to be the case with the offsets I described in the original post. However, I have debugged a bit further since then.

From what I can see the `createAMDRegister` function is called twice, and on the second call it only supplies the deps `['@angular/common', '@angular/core']`. The first call is caused by the `define` call, and the second is triggered by a call to `getRegister` in the `amd.js` file (caused by an `instantiate` call).

My code to load the specific UMD bundle is a simple `load` call.

**Comment by guybedford** [comment]
Ahh, then it may be a mutation bug then.

I've added a possible fix in https://github.com/systemjs/systemjs/commit/0ec9da6b46d38da3813357124202b4c43da729d6. Perhaps try that out and let me know if it works for you?

**Comment by KinsheX** [comment]
That seems to do the trick - it works.

Though if I might comment on the possible fix, is there any reason you are using `[].concat(deps)` rather than `deps.slice(0)` to take a copy of the array?

**Comment by guybedford** [comment]
Great to hear. Will add this to the next release then.

Nope, just old habits.

**Comment by KinsheX** [comment]
Perfect. Thank you for the swift handling.

**Comment by KinsheX** [comment]
*Hit the wrong button

**Comment by guybedford** [comment]
Released in 3.1.4.
